### PR TITLE
Add Softplus activation function @open sesame 04/15 10:41

### PR DESCRIPTION
**Pull request by heka1024**
- Now, user can use Softplus activation function like torch or tensor flow.
- Furthermore, we can use this function to build Mish or other activation functions

**Self evaluation:**
1. Build test:     [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: heka1024 <heka1024@gmail.com>

### CODE REVIEW

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by djeong20 — APPROVED** [review]
LGTM

File: nntrainer/layers/acti_func.h [hunk 1]
```diff
@@ -344,6 +347,27 @@ class ActiFunc {
...
   }
 
+  /**
+   * @brief     Softplus activation function
+   * @tparam T type of an input/output
+   * @param x input
+   * @return T type output
+   */
+  template <typename T = float> static T softplus(T x) {
+    /** TODO: Change beta to be a property */
```
Comment (jijoongmoon) [inline nntrainer/layers/acti_func.h:357]: For now, It is OK. but we need to have proper API for acti_function to have properties.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Comment by myungjoo** [comment]
Please rebase:
```
$ git checkout main
$ git pull ${FILL_upstream_name} main
$ git checkout activation-softplus
$ git rebase main
$ git push ${FILL_your_fork_name} HEAD -f
```

**Review by myungjoo — APPROVED** [review]

**Comment by skykongkong8** [comment]
I found this PR raises error when building with
`meson build -Denable-fp16=true`
@heka1024 
Could you please take a look?

**Comment by skykongkong8** [comment]
@heka1024 Nope it's build fail, not unittest fail.  

```bash
 error: converting to ‘_Float16’ from ‘float’ with greater conversion rank [-Werror]
  364 |     return static_cast<T>(log(1 + exp_util<T>(beta * x)) / beta);
```
```bash
 error: converting to ‘_Float16’ from ‘float’ with greater conversion rank [-Werror]
  374 |     return sigmoid<T>(beta * x);
      |            ~~~~~~~~~~^~~~~~~~~~
../nntrainer/layers/acti_func.h:269:52: note:   initializing argument 1 of ‘static T nntrainer::ActiFunc::sigmoid(T) [with T = _Float16]’
  269 |   template <typename T = float> static T sigmoid(T x) {
      |  
```

Try below to generate the error above:
```
cd ~/nntrainer/
meson build -Denable-fp16=true
```

Think your commit needs some fixes. Could you create a new PR to fix this issue?

**Comment by skykongkong8** [comment]
@heka1024
This issue is resolved via #2557
I would like to personally recommend you to check for fp16 build case as well next time!

**Pull request merged**